### ISSUE TITLE
search cards do not accurately reflect ability to download file (permission inheritance)

### ISSUE THREAD

**Issue by pdurbin** [IQSS/dataverse#4387]
While working on #4268 I observed some strange behavior that I demo'ed to @dlmurphy this morning having to do with inheritance of the "File Download" permission. Our expectation is that if you assign the "File Downloader" permission at the dataverse level or dataset level that the user will be able to download all restricted files. That's even what the UI says:

> "Assigning the File Downloader role means the user(s) will also have the File Downloader role applied to all Files within this Dataverse.

Here's a screenshot of the text above and one user giving the "File Downloader" role to another user at the dataverse level:

![screen shot 2017-12-15 at 1 11 21 pm](https://user-images.githubusercontent.com/21006/34054685-09addab8-e19a-11e7-9b58-d53db111ae7c.png)

The text above is accurate in that the user who now has the File Downloader role (Some Guy in the screenshots below) **can** download restricted files. The file appears with the the green "unlocked" icon on both the dataset and file pages...

## dataset page (green unlocked icon)

![screen shot 2017-12-15 at 1 12 15 pm](https://user-images.githubusercontent.com/21006/34054786-66db797a-e19a-11e7-9869-6c7e4a0c5d58.png)

## file page (green unlocked icon)

![screen shot 2017-12-15 at 1 12 19 pm](https://user-images.githubusercontent.com/21006/34054785-66d0507c-e19a-11e7-9082-346c4bf69322.png)

... but the "search card" mistakenly shows the red "locked" icon:

## search card (red locked icon)

![screen shot 2017-12-15 at 1 12 03 pm](https://user-images.githubusercontent.com/21006/34054788-66e707cc-e19a-11e7-930d-345f06d2b132.png)

The work around is to assign the "File Downloader" permission at the file level. This works because the code is checking on the DvObject itself...

`if ( permissionService.on(dvo).has(Permission.DownloadFile) ){`

... at https://github.com/IQSS/dataverse/blob/v4.8.4/src/main/java/edu/harvard/iq/dataverse/PermissionsWrapper.java#L201

In short, the work around can be used because inheritance of File Dowloader permission is not done in the code that decides whether to show the green or red icon on the search cards.

I'm able to demonstrate this bug with explicit groups as well. All testing was done on https://demo.dataverse.org running Dataverse 4.8.4

**Comment by mheppler** [comment]
If it's still broken, this should remain open until it is fixed. We have 838 other issues that remain open until fixed, why can't this one?

**Comment by pdurbin** [comment]
Everyone who opens GitHub issues, not just here but on any project, should reflect from time to time on if they really care if the issue gets fixed or not. Last night I updated my "GitHub Issues By Username" spreadsheet at https://docs.google.com/spreadsheets/d/110Y2ZG-q0ji8emo9ILzJe2jjqXoArTZB_Zt5J1ELiWg/edit?usp=sharing and looked at all the issues I had created. I closed the ones I don't really care about. Now there are only 5 open in my name (and 304 issues I have opened that are now closed) and they're all real bugs that I'd like to see fixed. Especially for people who have a lot of open issues (but everyone, really), I encourage you to look through the issues you opened and decide if you think having them open is adding a lot of value or not. Here's a list of people who have 10 or more open issues in their name:

![screen shot 2018-06-28 at 8 45 49 am](https://user-images.githubusercontent.com/21006/42035199-5d426090-7ab0-11e8-8e0b-de7aa79fffb1.png)

Issues have a tendency to come back. If this bug is driving someone crazy, they'll open a new one.

We need a better strategy for being responsive to open issues. 

"The tracker is as much a public face of the project as the repository, mailing lists or web pages." http://producingoss.com/en/producingoss.html#bug-tracker

Please see also the "issue backlog" thread I started a while ago: https://groups.google.com/d/msg/dataverse-dev/Pkces_MBqR8/Qsp7zrgfBgAJ
